WordPress/gutenberg
# wp-env: Add multisite support

closes #27961 

Add multisite support as an option to wp-env.

```json
{
	"env": {
		"development": {
			"multisite": true
```

<img width="526" alt="wp-env-multisite" src="https://github.com/user-attachments/assets/c5054fb7-9a23-4868-89a3-08e821e56120" />

## Why?
Makes it easier for developers to bootstrap multisite environments, especially for ease of feature testing and quality control.

## How?
* Adds a .wp-env.json environment config key (`multisite`) in the form of a boolean, defaulting to false.
* During wp-env's install and configuration phase, as part of the `start` command, if the environment has the `multisite` flag, then:
  * Instead of instructing the `cli` container to run `wp core install` on the environment's `wordpress` container, run [`wp core multisite-install`](https://developer.wordpress.org/cli/commands/core/multisite-install/).
  * Immediately thereafter, create an `.htaccess` file in the environment's `wordpress` container containing the rewrite rules required by the multisite setup.

## Testing Instructions
_Remember to `npx wp-env destroy` in between tests._

* Start a new wp-env setup (e.g. in a new directory) with no modifications. Verify that the result is a single-site setup, as it's always been.
* Start a new wp-env setup whose `.wp-env.json` sets the multisite flag to true.
  * Verify that the result is a multisite network.
  * Verify that you can create new sites (via `/wp-admin/network/sites.php`).
  * Verify that you can visit the backend and frontend of such new sites (e.g. http://localhost:8888/foo/).
  * Verify that you can visit the backend and frontend of the main site.
